org-roam-db.el

```diff
@@ -319,7 +319,7 @@ Insertions can fail if the key is already in the database."
 If the file does not have any connections, nil is returned."
   (let* ((query "WITH RECURSIVE
                    links_of(file, link) AS
-                     (WITH filelinks AS (SELECT * FROM links WHERE \"type\" = '\"file\"'),
+                     (WITH filelinks AS (SELECT * FROM links WHERE NOT \"type\" = '\"cite\"'),
```

Comment:
That question is wrt to this query, where `filelinks` will then contain all sorts of other links (including id links).

I'm just curious as to what additional link types you needed support for.

Reply:
Any types defined in Org, actually. I can then show them differently on the graph, for example.

But what I need first are types I myself can define. Semantically, notes in Org-roam can be connected differently. A link can denote the continuation of a main theme, a sort of  detalization, or simple informational link, or whatever. Luhman in his Zettels differentiated them through numbering. So I can, for example, define linktypes `major:` and `minor:` and see the difference in connections visually if I adjust faces or colors on the graph. At the moment, I'm not sure what linktypes I (or other users) might need and I'd like to have some general mechanism for that.

By the way, I noticed an error in handling id links, I'll need to check it.

Reply:
> So I can, for example, define linktypes major: and minor: and see the difference in connections visually if I adjust faces or colors on the graph.

So my only concern with this now is that Org-roam isn't aware of what kind of link that is supposed to be. Is the path a file path, or an id, or something else entirely? Org-roam doesn't handle this well now, so this PR is only part of the solution.

Reply:
Yes, and this is why in the beginning I introduced the concept of "inner link types" which Org-roam knows how to handle. But I don't think this is a problem, because when Org-roam doesn't know the type, it passes it to Org (right?), which knows it. As for custom links, users should describe how to handle them. For example, I define `major:` and `minor:` links as virtually equal to `file:`, only faces are different.

Reply:
There is a problematic place though: `org-roam--expand-links` should expand only `file:`-like links. Or. maybe, only links, pointing to Org-roam files.

Reply:
Let's keep `org-roam--expand-links` such that it expands only `file:` links for now, and then in a later PR, we can introduce a `org-roam-link-types` alist that acts as a lookup so the different link types can be handled appropriately.

Reply:
What if we expand any link type but only if the path belongs to the org-roam-directory? This makes sense.

Reply:
This is fine too.

Reply:
ok, done.

Reply:
Tests failed by timeout.

Reply:
It looks like that last change made the cache build much slower.